core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -91,7 +91,7 @@ class LogSegment private[log] (val log: FileRecords,
   private var bytesSinceLastIndexEntry = 0
 
   /* The timestamp we used for time based log rolling */
-  private var rollingBasedTimestamp: Option[Long] = None
+  @volatile private var rollingBasedTimestamp: Option[Long] = None
```

Comment:
can you add a comment stating why this needs to be volatile.

Reply:
added